### PR TITLE
OCI GenAI

### PR DESCRIPTION
**Pull request by anders-swanson**

### CODE REVIEW

File: spring-cloud-oci-gen-ai/pom.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright (c) 2023, Oracle and/or its affiliates.
```
Comment (markxnelson) [inline spring-cloud-oci-gen-ai/pom.xml:3]: 2024

File: spring-cloud-oci-samples/spring-cloud-oci-gen-ai-sample/pom.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright (c) 2023, Oracle and/or its affiliates.
```
Comment (markxnelson) [inline spring-cloud-oci-samples/spring-cloud-oci-gen-ai-sample/pom.xml:3]: 2024

**Review by markxnelson — DISMISSED** [review]
did a quick review, and looks good.  have not done an exhaustive review yet. 
a few notices wrong format or year

File: spring-cloud-oci-samples/spring-cloud-oci-gen-ai-sample/src/test/resources/application-chat-cohere.properties [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+
+spring.cloud.oci.genai.chat.on-demand-model-id=ocid1.generativeaimodel.oc1.us-chicago-1.amaaaaaask7dceyawk6mgunzodenakhkuwxanvt6wo3jcpf72ln52dymk4wq
```
Comment (andytael) [inline spring-cloud-oci-samples/spring-cloud-oci-gen-ai-sample/src/test/resources/application-chat-cohere.properties:4]: I don't think we can publish OCIDs on GitHub

File: spring-cloud-oci-samples/spring-cloud-oci-gen-ai-sample/src/test/resources/application-chat-llama.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+
+spring.cloud.oci.genai.chat.on-demand-model-id=ocid1.generativeaimodel.oc1.us-chicago-1.amaaaaaask7dceyaycmwwnvu2gaqrffquofgmshlqzcdwpk727n4cykg34oa
```
Comment (andytael) [inline spring-cloud-oci-samples/spring-cloud-oci-gen-ai-sample/src/test/resources/application-chat-llama.properties:4]: I don't think we can publish OCIDs on GitHub

**Comment by anders-swanson** [comment]
Added documentation and updated for review comments.

**Review by andytael — APPROVED** [review]
LGTM

**Pull request merged**